File: java-engine/src/main/java/hu/elteik/knowledgelab/javaengine/core/models/rotorrouterdispersion/RotorRouterDispersionNode.java
package hu.elteik.knowledgelab.javaengine.core.models.rotorrouterdispersion;

import hu.elteik.knowledgelab.javaengine.core.models.base.Color;
import hu.elteik.knowledgelab.javaengine.core.models.base.Node;
import hu.elteik.knowledgelab.javaengine.core.models.base.NodeState;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Map;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class RotorRouterDispersionNode extends Node {

    private Map<Color, Long> currentComponentPointer;

    public RotorRouterDispersionNode(Long ID, NodeState state, Map<Color, Long> currentComponentPointer) {
        super(ID, state);
        this.currentComponentPointer = currentComponentPointer;
    }
}

File: dispersion-app/src/app/models/algorithms/rotor-router-dispersion/RotorRouterDispersionNode.ts
import { Node } from '../../core/Node';
import { NodeState } from '../../utils/NodeState';

export class RotorRouterDispersionNode extends Node {
  rotorRouter: number;

  constructor(id: number, state: NodeState, rotorRouter: number) {
    super(id, state);
    this.rotorRouter = rotorRouter;
  }
}
